tests/acceptance/course-page/start-course-test.js

```diff
@@ -17,6 +17,33 @@ module('Acceptance | course-page | start-course', function (hooks) {
   setupApplicationTest(hooks);
   setupAnimationTest(hooks);
 
+  test('can auto select language coming from track page', async function (assert) {
+    testScenario(this.server, ['dummy']);
+    signIn(this.owner, this.server);
+
+    const course = this.server.schema.courses.findBy({ slug: 'dummy' });
+    course.update({ releaseStatus: 'live' });
+
+    await catalogPage.visit();
+    await catalogPage.clickOnTrack('Python');
+    await trackPage.clickOnCourseCard('Build your own Dummy →');
+
+    await percySnapshot('Auto Select Language - Before Clicking Show-Other-Languages Button');
+
+    assert.strictEqual(coursePage.createRepositoryCard.expandedSectionTitle, 'Preferred Language', 'current section title is preferred language');
+    assert.strictEqual(coursePage.createRepositoryCard.languageButtons.length, 1, 'only one language-button can be visible');
+    assert.strictEqual(coursePage.createRepositoryCard.languageButtons[0].text, 'Python', 'the only one language-button must be Python');
+    assert.ok(coursePage.createRepositoryCard.showOtherLanguagesButton.isVisible, 'show-other-languages button is visible');
+
+    await coursePage.createRepositoryCard.clickOnShowOtherLanguagesButton();
```

Comment:
Btw, if you have `showOtherLanguagesButton` already defined, you can just do `showOtherLanguagesButton.click()` here and don't need to define the extra `clickOnShowOtherLanguagesButton` helper.

Reply:
Thx for the tip, Paul! I just tried that, but it's somehow not working.

`showOtherLanguagesButton` seems to be disappearing if not used with `assert`. For example, I put these three lines right next to each other (just for testing purpuoses):

```javascript
console.log(coursePage.createRepositoryCard.ShowOtherLanguagesButton);
assert.ok(coursePage.createRepositoryCard.showOtherLanguagesButton.isVisible, 'show-other-languages button is visible');
await coursePage.createRepositoryCard.ShowOtherLanguagesButton.click();
```

If `assert.ok` works, I don't expect the 1st and 3rd lines not to work. However,

```javascript
console.log(coursePage.createRepositoryCard.ShowOtherLanguagesButton);
// logging out undefined

assert.ok(coursePage.createRepositoryCard.showOtherLanguagesButton.isVisible, 'show-other-languages button is visible');
// this works

await coursePage.createRepositoryCard.ShowOtherLanguagesButton.click();
// TypeError: Cannot read properties of undefined (reading 'click')
```

<img width="805" alt="image" src="https://github.com/user-attachments/assets/e4889da7-b6fc-471b-8e62-b39290b71e5f">